python/ray/ml/config.py

```diff
@@ -35,29 +34,71 @@ class ScalingConfigDataClass:
         Strategies <pgroup-strategy>` for the possible options.
     """
 
-    num_workers: int
+    num_workers: int = 1
```

Comment:
Can we make this a required arg instead? Not sure if 1 is always a sensible default.

Reply:
Since we decided to have a single ScalingConfig for all Trainers, then I think it only makes sense to have a required arg if it should be required for all Trainers.

In this case, num_workers only makes sense if your Trainer is creating new actors. For the general case, `num_workers` doesn't mean anything, so then I don't think we should make it required.

Reply:
Hmm but by that logic wouldn't it make less sense if `num_workers=1` for the general case?

Reply:
Changed!